### ISSUE TITLE
[Usage] OOM  for 7B LLAVA v1.5 finetuning 8xA100(40G)

### ISSUE THREAD

**Issue by CHENGY12** [haotian-liu/LLaVA#655]
### Describe the issue

Issue:
I tried to use 2 nodes with 8xA100(40G)s to finetune the 7B models. But it still reports OOM error.  

Command:
```
srun --gres=gpu:4 --ntasks-per-node=1 --nodes=2 --cpus-per-task=128 --mem=160G deepspeed llava/train/train_mem.py \
    --deepspeed ./scripts/zero2.json \
    --model_name_or_path lmsys/vicuna-7b-v1.5 \
    --version v1 \
    --data_path ./playground/data/llava_v1_5_mix665k.json \
    --image_folder ./playground/data \
    --vision_tower openai/clip-vit-large-patch14-336 \
    --pretrain_mm_mlp_adapter ./checkpoints/llava-v1.5-7b-pretrain/mm_projector.bin \
    --mm_projector_type mlp2x_gelu \
    --mm_vision_select_layer -2 \
    --mm_use_im_start_end False \
    --mm_use_im_patch_token False \
    --image_aspect_ratio pad \
    --group_by_modality_length True \
    --bf16 True \
    --output_dir ./checkpoints/llava-v1.5-7b-finetune \
    --num_train_epochs 1 \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 1 \
    --gradient_accumulation_steps 16 \
    --evaluation_strategy "no" \
    --save_strategy "steps" \
    --save_steps 10000 \
    --save_total_limit 10 \
    --learning_rate 2e-5 \
    --weight_decay 0. \
    --warmup_ratio 0.03 \
    --lr_scheduler_type "cosine" \
    --logging_steps 1 \
    --tf32 True \
    --model_max_length 2048 \
    --gradient_checkpointing True \
    --dataloader_num_workers 4 \
    --lazy_preprocess True \
    --report_to wandb
```

**Comment by Rickylht** [comment]
try zero3+gradient_accumulation_steps = 1

**Comment by haotian-liu** [comment]
If you have 8x A100 (40G), you should be able to run it with batch size 16, gradient accu 1, under zero3.json

**Comment by microhu** [comment]
> If you have 8x A100 (40G), you should be able to run it with batch size 16, gradient accu 1, under zero3.json

how to estimate the required memory and. better set the batchhsize? I changed the batchsize from 1 to 16, the gpu memory didn't changed too much when gradient_checkpoint is enabled in finetuning stage.

**Comment by CHENGY12** [comment]
I have tried for zero3 too.  It still shows the OOM issue. We used 
  --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 1 \
    --gradient_accumulation_steps 16 \
    
      --per_device_train_batch_size 16 \
    --per_device_eval_batch_size 1 \
    --gradient_accumulation_steps 1 \
    
    It doesn't work.  I will check with --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 1 \
    --gradient_accumulation_steps 1 \
    later.

**Comment by haotian-liu** [comment]
Hi @CHENGY12

We just released LoRA finetuning scripts and checkpoints, and with LoRA, it should be able to enough to fit even 13B training into 8-A100-40g.

https://github.com/haotian-liu/LLaVA/releases/tag/v1.1.3

Let me know if you still have any issues.

**Comment by CHENGY12** [comment]
Thanks for your kind updates.  I will try the LORA!

**Comment by LumenYoung** [comment]
> Hi @CHENGY12
> 
> We just released LoRA finetuning scripts and checkpoints, and with LoRA, it should be able to enough to fit even 13B training into 8-A100-40g.
> 
> https://github.com/haotian-liu/LLaVA/releases/tag/v1.1.3
> 
> Let me know if you still have any issues.

Dear Haotian Liu,

Thanks for the lora checkpoint and I have a question regarding the computation resource for finetuning lora models. I currently have 24 GBs GPU vram. It is possible to use such resource to finetune 13B model with smaller batchsizes?

Thanks in advance for explaining.
